spec/factories/trainees.rb

```diff
@@ -0,0 +1,17 @@
+FactoryBot.define do
+  factory :trainee do
+  end
+
+  factory :trainee_for_form, class: Trainee do
```

Comment:
I'd say just keep this in the main `:trainee` factory and then use traits to add variations. That way you can build up the fixture as you need it. ie instead of using `create(:trainee_for_form)` -> `create(:trainee, :with_form_details)`

Reply:
yea i tried that, didn't work due to the way it inherits. eg the form version doesn't have a `date_of_birth` field. If i put `date_of_birth` in the parent/base factory, the trait will inherit. I don't think i can remove it unless i start doing some quirky stuff